### PR TITLE
Use webpack / fix sass in dev

### PR DESCRIPTION
**Pull request by sndrs**
## What does this change?

- fixes the `sass` compiling/loading endless loop (@superfrank, @zeftilldeath)
- centralises `css` creation (`sass`/`postcss` etc)
- uses `webpack` in `dev`
  - make `webpack` config configurable by env
- removes `gulp`
- clears out some other unused packages from `package.json`

## What is the value of this and can you measure success?

- with `webpack` in `dev` we can confidently remove `curl` in `prod` (no unexpected surprises in `prod`)
- `sass` works properly in `dev` again
- `dev` `css` and `prod` `css` are the same (no unexpected surprises in `prod`)
- provides sourcemaps for `css` and `js` in `dev` and `prod`

## Tested on CODE?
- [x] test on CODE

### CODE REVIEW

**Comment by superfrank** [comment]
image

**Comment by SiAdcock** [comment]
And maybe the sourcemaps will work in Sentry too!

**Comment by sndrs** [comment]
yeah, fingers crossed...

File: dev/watch.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+const webpack = require('webpack');
+const webpackBundler = webpack(require('../webpack.config.js')());
+
+webpackBundler.watch({
+    ignored: /node_modules/,
+}, (err, stats) => {
+    if (err) {
+        // log any unexpected error
+        console.err('webpack error:', err);
```
Comment (SiAdcock) [inline dev/watch.js]: Would be cool to implement [a nice logging module](https://github.com/jonnyreeves/js-logger) one day!

File: dev/watch.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    if (INITIAL_BUNDLE) {
+        INITIAL_BUNDLE = false;
+        wpNotification.succeed();
+        console.log('');
```
Comment (SiAdcock) [inline dev/watch.js]: What is this `console.log` doing?

Comment (sndrs) [inline dev/watch.js]: just adds a line of white space, i'll add a comment

File: dev/watch.js (hunk 1 above)
Comment (sndrs) [inline dev/watch.js]: i really wish node did that by default

File: dev/watch.js (hunk 2 above)
Comment (sndrs) [inline dev/watch.js]: oh no, i think it's old, good spot

**Comment by SiAdcock** [comment]
Really tidy, man. Love the code sharing between the watch script and the compile tool.

**Review by gtrufitt — APPROVED** [review]

**Pull request merged**